File: 2779-maximum-beauty-of-an-array-after-applying-operation/2779-maximum-beauty-of-an-array-after-applying-operation.py
class Solution:
    def maximumBeauty(self, nums: list[int], k: int) -> int:
        mx = max(nums)
        vis = [0] * (mx + 2)
        for x in nums:
            vis[max(0, x - k)] += 1
            vis[min(x + k + 1, mx + 1)] -= 1

        ans, cnt = 0, 0
        for x in vis:
            cnt += x
            ans = max(ans, cnt)

        return ans

File: 2779-maximum-beauty-of-an-array-after-applying-operation/2779-maximum-beauty-of-an-array-after-applying-operation.cpp
class Solution {
public:
    
    int maximumBeauty(vector<int>& nums, int k) {
      

        int mx = *max_element(nums.begin(),nums.end());

        vector<int>vis(mx+1);
        int n = nums.size();
        for(int x : nums){
            vis[max(0,x-k)]++;
            vis[min(x+k+1,mx)]--;
        }
        int ans = 1,cnt = 0;
        for(int x : vis){
            cnt+=x;
            ans=max(ans,cnt);
        }
        return ans;

    }
};